pkg/health/checker/checker.go

```diff
@@ -265,23 +345,110 @@ func (c consulHealthChecker) WatchService(
 				case resultCh <- out:
 				}
 			}
+			if useHealthService {
```

Comment:
you'll also be able to get rid of this `if` if you use different types for the shadow and not-shadow checker